File: software/examples/javascript/ExampleEdgeCount.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'XYZ'; // Change XYZ to the UID of your Industrial Digital In 4 Bricklet 2.0

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var idi4 = new Tinkerforge.BrickletIndustrialDigitalIn4V2(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function (error) {
        console.log('Error: ' + error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

ipcon.on(Tinkerforge.IPConnection.CALLBACK_CONNECTED,
    function (connectReason) {
        // Configure rising edge count (channel 3) with 10ms debounce
        idi4.setEdgeCountConfiguration(3, 0, 10);

        // Get edge count 10 times with 1s delay
        for(var i = 0; i < 10; ++i) {
            setTimeout(function () {
                // Get current edge count
                idi4.getEdgeCount(3, false,
                    function (count) {
                        console.log('Count: ' + count);
                    },
                    function (error) {
                        console.log('Error: ' + error);
                    }
                );
            }, 1000 * i + 1000);
        }
    }
);

console.log('Press key to exit');
process.stdin.on('data',
    function (data) {
        ipcon.disconnect();
        process.exit(0);
    }
);

File: software/examples/rust/example_edge_count.rs
use std::{io, error::Error};
use std::thread;
use std::time::Duration;
use tinkerforge::{ip_connection::IpConnection, 
                  industrial_digital_in_4_v2_bricklet::*};


const HOST: &str = "localhost";
const PORT: u16 = 4223;
const UID: &str = "XYZ"; // Change XYZ to the UID of your Industrial Digital In 4 Bricklet 2.0.

fn main() -> Result<(), Box<dyn Error>> {
    let ipcon = IpConnection::new(); // Create IP connection.
    let idi4 = IndustrialDigitalIn4V2Bricklet::new(UID, &ipcon); // Create device object.

    ipcon.connect((HOST, PORT)).recv()??; // Connect to brickd.
    // Don't use device before ipcon is connected.

		// Configure rising edge count (channel 3) with 10ms debounce
		idi4.set_edge_count_configuration(3, 0, 10);

		// Get edge count 10 times with 1s delay
		for i in 0..10{
			thread::sleep(Duration::from_millis(1000));

			// Get current edge count.
	let count = idi4.get_edge_count(3, false).recv()?;
			println!("Count: {}", count);
		}

    println!("Press enter to exit.");
    let mut _input = String::new();
    io::stdin().read_line(&mut _input)?;
    ipcon.disconnect();
    Ok(())
}
